src/Accounts/Authentication/Sanitizer/DefaultProviderResolver.cs

```diff
@@ -0,0 +1,214 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.WindowsAzure.Commands.Common.Sanitizer
+{
+    public class DefaultProviderResolver : ISanitizerProviderResolver
+    {
+        private readonly SanitizerProviderCache<Type, SanitizerProviderBase> _providerCache;
+
+        private static readonly ISanitizerProviderResolver _instance = new DefaultProviderResolver();
+
+        public static ISanitizerProviderResolver Instance => _instance;
+
+        public ISanitizerService Service => AzureSession.Instance.TryGetComponent<ISanitizerService>(nameof(ISanitizerService), out var service) ? service : null;
```

Comment:
Initialize DefaultSanitizerService directly instead of via TryComponent. The code has been moved out of common.